src/Passes/Source/RuleTransformationPass/RulePass.cpp

```diff
@@ -0,0 +1,863 @@
+                 if (type == nullptr)
+                 {
+                     return false;
+                 }
+
+                 if (branch_cond)
+                 {
+                     builder.CreateBr(if_true);
+                     instr->replaceAllUsesWith(llvm::UndefValue::get(type));
+                 }
+                 else
+                 {
+                     builder.CreateBr(if_false);
+                     instr->replaceAllUsesWith(llvm::UndefValue::get(type));
+                 }
+
+                 replacements.push_back({val, nullptr});
+
+                 return true;
+             }});
+
+        if (config_.assumeNoExceptions())
+        {
+            // Replacing all invokes with calls
+            addConstExprRule({unnamedInvoke(), [](Builder& builder, Value* val, Captures&, Replacements& replacements) {
+                                  auto invoke = llvm::dyn_cast<llvm::InvokeInst>(val);
+                                  if (invoke == nullptr)
+                                  {
+                                      return false;
+                                  }
+
+                                  auto callee_function = invoke->getCalledFunction();
+
+                                  // List with new call arguments
+                                  std::vector<llvm::Value*> new_arguments;
+                                  for (unsigned i = 0; i < invoke->getNumArgOperands(); ++i)
+                                  {
+                                      // Getting the i'th argument
+                                      llvm::Value* arg = invoke->getArgOperand(i);
+
+                                      new_arguments.push_back(arg);
+                                  }
+
+                                  // Creating a new call
+                                  auto* new_call = builder.CreateCall(callee_function, new_arguments);
+                                  builder.CreateBr(invoke->getNormalDest());
+
+                                  new_call->takeName(invoke);
+
+                                  // Replace all calls to old function with calls to new function
+                                  invoke->replaceAllUsesWith(new_call);
+                                  replacements.push_back({invoke, nullptr});
+                                  return true;
+                              }});
+        }
+    }
+
+    void RuleTransformationPass::addConstExprRule(ReplacementRule&& rule)
+    {
+        auto ret = std::make_shared<ReplacementRule>(std::move(rule));
+
+        const_expr_replacements_.addRule(ret);
+    }
+
+    void RuleTransformationPass::constantFoldFunction(llvm::Function& function)
+    {
+        std::vector<llvm::Instruction*> to_delete;
+
+        // Folding all constants
+        for (auto& basic_block : function)
+        {
+
+            for (auto& instr : basic_block)
+            {
+                auto        module = instr.getModule();
```

Comment:
[`module`](https://en.cppreference.com/w/cpp/language/modules) is a C++ keyword starting with C++20. Consider renaming.
(VSCode has highlighted it as a keyword, so you may want to run through your code with the VSCode)

Reply:
Thanks - I was not aware of this. I would suggest that we put this on the backlog as this will be a larger task since the word `module` is used a lot of places and unfortunately, it is not just a matter of a simple "search and replace" as it would break a bunch of functions.

Reply:
Up to you, but I would recommend to rename, since the issue can show itself upon next migration to the new compiler version.
(In VSCode you can place a text cursor on the word, press <F2> (rename), enter the new name, and it will give you the _preview_ of all the changes (including in the comments). You can disable some of those changes and then run the remaining (enabled) changes in one click)

Reply:
I agree that this needs to be refactored. @bettinaheim just to make sure that you are also aware of this, we should record this in the backlog. While this is not an immediate task (we are using C++14), it would be good to sort this out as it otherwise may become a headache later on.